tidy3d/web/api/autograd/autograd.py

```diff
@@ -485,28 +487,13 @@ def _run_bwd(
     def vjp(data_fields_vjp: AutogradFieldMap) -> AutogradFieldMap:
         """dJ/d{sim.traced_fields()} as a function of Function of dJ/d{data.traced_fields()}"""
 
-        sim_adj = setup_adj(
+        sim_adj, adjoint_source_info = setup_adj(
             data_fields_vjp=data_fields_vjp,
             sim_data_orig=sim_data_orig,
             sim_data_fwd=sim_data_fwd,
             sim_fields_original=sim_fields_original,
         )
 
-        # no adjoint sources, no gradient for you :(
-        if not len(sim_adj.sources):
-            td.log.warning(
-                "No adjoint sources generated. "
-                "There is likely zero output in the data, or you have no traceable monitors. "
-                "As a result, the 'SimulationData' returned has no contribution to the gradient. "
-                "Skipping the adjoint simulation. "
-                "If this is unexpected, please double check the post-processing function to ensure "
-                "there is a path from the 'SimulationData' to the objective function return value."
-            )
-
-            # TODO: add a test for this
-            # construct a VJP of all zeros for all tracers in the original simulation
-            return {path: 0 * value for path, value in sim_fields_original.items()}
```

Comment:
What was the reason to add and now remove this? Like are you sure it cannot happen anymore?

Reply:
that's right.